src/webui/api/appcontroller.cpp

```diff
@@ -1083,6 +1085,34 @@ void AppController::defaultSavePathAction()
     setResult(BitTorrent::Session::instance()->savePath().toString());
 }
 
+void AppController::getDirectoryContentAction()
+{
+    requireParams({u"dirPath"_s, u"mode"_s});
+    const QString visibility = params().value(u"mode"_s);
+    if (visibility != u"all"_s && visibility != u"dirs"_s && visibility != u"files"_s)
```

Comment:
I would avoid testing visibility value twice:
```suggestion
    const QString visibility = params().value(u"mode"_s, u"all"_s);
    if (visibility == u"dirs")
        filters |= QDir::Dirs;
    else if (visibility == u"files")
        filters |= QDir::Files;
    else if (visibility == u"all")
        filters |= QDir::Dirs | QDir::Files;
        else
            throw APIError(APIErrorType::BadParams, u"Invalid mode, allowed values: all, dirs, files"_s);
```

Reply:
And even extract it into function:
```c++
QDir::Filters parseDirectoryContentMode(const QString &visibility)
{
    if (visibility == u"dirs")
        return QDir::Dirs;
    if (visibility == u"files")
        return QDir::Files;
    if (visibility == u"all")
        return (QDir::Dirs | QDir::Files);
    throw APIError(APIErrorType::BadParams, u"Invalid mode, allowed values: all, dirs, files"_s);
```

Reply:
Done

Reply:
Done